File: src/main/java/com/dozycoffee/wms/warehouse/domain/enumeration/AreaCode.java
package com.dozycoffee.wms.warehouse.domain.enumeration;

import com.dozycoffee.wms.warehouse.domain.valueobject.Capacity;
import lombok.Getter;

@Getter
public enum AreaCode {

    INBOUND("입고 처리장", new Capacity(50)),
    OUTBOUND("출고장", new Capacity(50)),
    RETURN("반품 처리장", new Capacity(30)),
    DISPOSAL("폐기 처리장", new Capacity(20));

    private final String areaName;
    private final Capacity capacity;

    AreaCode(String areaName, Capacity capacity) {
        this.areaName = areaName;
        this.capacity = capacity;
    }
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/domain/enumeration/AreaCode.kt
package com.dozycoffee.wms.warehouse.domain.enumeration

import com.dozycoffee.wms.warehouse.domain.valueobject.Capacity

enum class AreaCode(
    val areaName: String,
    val capacity: Capacity
) {
    INBOUND("입고 처리장", Capacity(50)),
    OUTBOUND("출고장", Capacity(50)),
    RETURN("반품 처리장", Capacity(30)),
    DISPOSAL("폐기 처리장", Capacity(20))
}
